### PR TITLE
DFPL-2148: Migration to remove documents from case

### PR DESCRIPTION
**Pull request by qasimnawaz-moj**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DFPL-2148
https://tools.hmcts.net/jira/browse/DFPL-2149

### Change description ###

Migration to remove documents from case

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseServiceTest.java [hunk 1]
```diff
@@ -927,6 +929,47 @@ void shouldThrowExceptionIfStandardDirectionOrderNotMatching() {
...
+                .id(caseId)
+                .returnApplication(
+                    ReturnApplication.builder()
+                        .document(documentReference)
+                        .build())
+                .build();
+
+            assertThrows(AssertionError.class, () -> underTest
+                .verifyReturnApplicationExists(caseData, MIGRATION_ID, document1Id));
+        }
```
Comment (so99y) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseServiceTest.java:970]: Missing a test case for not throwing error

**Review by so99y — COMMENTED** [review]
please take a look on the comments.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java [hunk 2]
```diff
@@ -609,6 +628,31 @@ public Map<String, Object> removeNoticeOfProceedingsBundle(CaseData caseData, St
...
+            ElementUtils.removeElementWithUUID(documentsFiledOnIssueList, documentFiledOnIssueId);
+
+        if (updatedDocumentsFiledOnIssueList.size() != documentsFiledOnIssueList.size() - 1) {
+            throw new AssertionError(format(
+                "Migration {id = %s, case reference = %s}, document filed on issue %s not found",
+                migrationId, caseId, documentFiledOnIssueId
+            ));
+        }
+
+        return Map.of("documentsFiledOnIssueList", updatedDocumentsFiledOnIssueList);
```
[the remark this answers is not in the document]
Comment (qasimnawaz-moj) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java:653]: According to Neil, they can't so we need this migration

**Review by so99y — DISMISSED** [review]

**Review by AlistairEdwinOsborne — APPROVED** [review]
all good!

**Pull request merged**